browser_patches/firefox/juggler/NetworkObserver.js

```diff
@@ -684,6 +752,8 @@ class NetworkObserver {
   _onRequest(channel, topic) {
     if (!(channel instanceof Ci.nsIHttpChannel))
       return;
+    if (this._responseInterceptionChannels.has(channel))
```

Comment:
I think you should also ignore this in more places, e.g. `_onRedirect`or `_onResponse`? Perhaps we need a test for intercepted response after a redirect?

Reply:
`_onResponse` will ignore the channel since it is not in `this._channelToRequest` map (skipped in `_onRequest`, the redirects chain maybe more tricky though, I'll add a test.

Reply:
Response interception is broken one way or the other in all 3 browsers in case of redirects so I'll address it separately.